WooCommerce/Classes/ViewRelated/Orders/Order Details/Receipts/ReceiptViewController.swift

```diff
@@ -4,11 +4,19 @@ import WebKit
 final class ReceiptViewController: UIViewController {
     @IBOutlet private weak var webView: WKWebView!
     private let viewModel: ReceiptViewModel
+    
+    var onDismiss: (() -> Void)?
 
-    init(viewModel: ReceiptViewModel) {
+    init(viewModel: ReceiptViewModel, onDismiss: (() -> Void)? = nil) {
         self.viewModel = viewModel
+        self.onDismiss = onDismiss
         super.init(nibName: "LegacyReceiptViewController", bundle: nil)
     }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        onDismiss?()
+    }
```

Comment:
Is this the best place to call `onDismiss`? 

It seems a little separate from any intention to dismiss the view, and could be caused by other non-dismiss related actions.

Perhaps the `onDismiss` handler should be renamed `onDisappear`, or it should be called from elsewhere (e.g. when a button is tapped to dismiss the receipt view.)

Reply:
At the moment the modal does not have a cancel button, but we might add it as well. The renaming sounds better indeed, specially if we end adding the button. Done on https://github.com/woocommerce/woocommerce-ios/pull/11896/commits/d2f6db639372840860d323988fe808529fbe1141